### PR TITLE
Add support for `/contacts/groups` endpoint

### PR DESCRIPTION
**Pull request by arunagw**
There is new endpoint for which we want to add support and it is `/contacts/groups`
This new endpoint is used to get all the `groups` from an account. Only list is allowed for the contact groups end point.

* Make `ContactGroup` class to be a model so we can fetch it standalone via api.
* Add spec to make sure we have all the attributes we want from the API.

### CODE REVIEW

**Comment by arunagw** [comment]
This PR also requested here https://github.com/nylas/nylas-ruby/issues/200

File: lib/nylas/contact_group.rb [hunk 1]
```diff
@@ -4,7 +4,15 @@ module Nylas
...
   class ContactGroup
-    include Model::Attributable
+    include Model
+    self.resources_path = "/contacts/groups"
+    self.creatable = false
+    self.listable = true
+    self.showable = false
+    self.filterable = false
+    self.updatable = false
+    self.destroyable = false
```
Comment (adarsh) [inline lib/nylas/contact_group.rb]: What do you think about sorting this list either alphabetically or by which are `true` then `false`?

File: spec/nylas/contact_group_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  it "is not destroyable" do
+    expect(described_class).not_to be_destroyable
+  end
+
+  it "is listable" do
+    expect(described_class).to be_listable
+  end
+
+  it "is not showable" do
+    expect(described_class).not_to be_showable
```
Comment (adarsh) [inline spec/nylas/contact_group_spec.rb]: I ❤️ these matchers

File: lib/nylas/contact_group.rb (hunk 1 above)
Comment (arunagw) [reply lib/nylas/contact_group.rb]: taken care here 1da3b949e049ad20a30dbd7287624e3084328207

**Pull request merged**